docs/design/experiment-implementation.md

```diff
@@ -0,0 +1,248 @@
+
+```
+experiment = create_experiment {
+    Environment = environment, 
+    ExperimentConfig = {
+       type = "adhoc",
+       localize_artifacts = [
+            "s3://bucket/training-job.tar.gz"
+       ],
+       name = "abc",
+       parameter = "python training.py --iteration 10 --input="s3://bucket/input output="s3://bucket/output",
+    }
+}
+experiment.run()
+experiment.wait_for_finish(print_output=True)
+```
+
+##### Run notebook file in offline mode
+
+It is possible we want to run a notebook file in offline mode, to do that, here's code to use to run a notebook code
+
+```
+experiment = create_experiment {
+    Environment = environment, 
+    ExperimentConfig = {
+       type = "adhoc",
+       localize_artifacts = [
+            "s3://bucket/folder/notebook-123.ipynb"
+       ],
+       name = "abc",
+       parameter = "runipy training.ipynb --iteration 10 --input="s3://bucket/input output="s3://bucket/output",
+    }
+}
+experiment.run()
+experiment.wait_for_finish(print_output=True)
+```
+
+##### Run pre-defined experiment library
+
+```
+experiment = create_experiment {
+    # Here you can use default environment of library
+    Environment = environment, 
+    ExperimentConfig = {
+       type = "template",
+       name = "abc",
+       # A unique name of template 
+       template = "deepfm_ctr", 
+       # yaml file defined what is the parameters need to be specified.
+       parameter = {
+           Input: "S3://.../input",
+           Output: "S3://.../output"
+           Training: {
+              "batch_size": 512,
+              "l2_reg": 0.01,
+              ...
+           }
+       }
+    }
+}
+experiment.run()
+experiment.wait_for_finish(print_output=True)
+```
+
+## Summarize: Experiment v.s. Notebook session
+
+There's a common misunderstanding about what is the differences between running experiment v.s. running task from a notebook session. We will talk about differences and commonalities:
+
+**Differences**
+
+|                                   | Experiment                                                   | Notebook Session                                             |
+| --------------------------------- | ------------------------------------------------------------ | ------------------------------------------------------------ |
+| Run mode                          | Offline                                                      | Interactive                                                  |
+| Output Artifacts (a.k.a model)    | Persisted in a shared storage (like S3/NFS)                  | Local in the notebook session container, could be emphameral |
+| Run history (meta, logs, metrics) | Meta/logs/metrics can be traced from experiment UI (or corresponding API) | No run history can be traced from Submarine UI/API           |
```

Comment:
After `No run history can be traced from Submarine UI/API`, add `Can view the current running log.`

Reply:
Updated